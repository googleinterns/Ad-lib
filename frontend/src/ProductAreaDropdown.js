import React from 'react';
import PropTypes from 'prop-types';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import ListItemText from '@material-ui/core/ListItemText';
import Select from '@material-ui/core/Select';

// TO-DO: Add more product area options to this array
const productAreas = [
  'Play',
  'Search',
  'Ads',
  'Youtube',
  'Maps',
];

// Add onChange to props validation
ProductAreaDropdown.propTypes = {
  onChange: PropTypes.func,
};

/**
 * Create product area dropdown component using productAreas array
 * @return {ProductAreaDropdown} ProductAreaDropdown component
 * @param {Object} props
 */
export default function ProductAreaDropdown(props) {
  return (
    <div>
      <FormControl style={{width: 180}}>
        <InputLabel id="productArea">Product Area</InputLabel>
        <Select
          id="productArea-input"
          name="productArea"
          onChange={(event) => props.onChange(event.target.value)}
          inputProps={{'aria-label': 'productArea'}}
        >
          {productAreas.map((currentProductArea) => (
            <MenuItem key={currentProductArea} value={currentProductArea}>
              <ListItemText primary={currentProductArea} />
            </MenuItem>
          ))}
        </Select>
      </FormControl>
    </div>
  );
}
