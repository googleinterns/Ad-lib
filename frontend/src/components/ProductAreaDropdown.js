// Copyright 2020 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React from 'react';
import PropTypes from 'prop-types';
import InputLabel from '@material-ui/core/InputLabel';
import {makeStyles} from '@material-ui/core/styles';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';

/** Establishes style to use on rendering component */
const useStyles = makeStyles((theme) => ({
  padding: {
    margin: theme.spacing(2),
    marginBottom: theme.spacing(3),
  },
}));

const productAreas = [
  'Ads',
  'Area 120',
  'Cloud',
  'Commerce',
  'Community Efforts',
  'Core',
  'Corporate Engineering',
  'Devices and Services',
  'Geo',
  'Global Affairs',
  'Global Business & Operations',
  'Global Communications & Public Affairs',
  'Google - advisors',
  'Google Finance',
  'Health',
  'Jigsaw',
  'Learning & Education',
  'Marketing',
  'Next Billion Users',
  'Payments',
  'People Operations',
  'Platforms & Ecosystems',
  'REWS (Real Estate & Workplace Services)',
  'Research',
  'Search',
  'Waze',
  'Youtube',
];

// Add onChange to props validation
ProductAreaDropdown.propTypes = {
  onChange: PropTypes.func,
};

/**
 * Create product area dropdown component using productAreas array
 * @param {Object} props
 * @return {ProductAreaDropdown} ProductAreaDropdown component
 */
export default function ProductAreaDropdown(props) {
  const classes = useStyles();

  return (
    <div>
      <FormControl style={{width: 180}}>
        <InputLabel id="productArea">Product Area</InputLabel>
        <Select
          id="productArea-input"
          name="productArea"
          onChange={(event) => props.onChange(event.target.value)}
          inputProps={{'aria-label': 'productArea'}}
        >
          {productAreas.map((currentProductArea) => (
            <option
              className={classes.padding}
              key={currentProductArea}
              value={currentProductArea}
            >
              {currentProductArea}
            </option>
          ))}
        </Select>
      </FormControl>
    </div>
  );
}
