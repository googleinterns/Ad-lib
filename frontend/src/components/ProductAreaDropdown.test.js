import React from 'react';
import ProductAreaDropdown from './ProductAreaDropdown';
import renderer from 'react-test-renderer';

describe('Product Area Dropdown', () => {
  it('should be defined', () => {
    expect(ProductAreaDropdown).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = renderer.create(<ProductAreaDropdown />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
