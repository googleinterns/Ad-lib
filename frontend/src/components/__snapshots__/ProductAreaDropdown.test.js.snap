// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Area Dropdown should render correctly 1`] = `
<div>
  <div
    className="MuiFormControl-root"
    style={
      Object {
        "width": 180,
      }
    }
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
      id="productArea"
    >
      Product Area
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        aria-label="productArea"
        aria-labelledby="productArea-input"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        id="productArea-input"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </div>
      <input
        aria-hidden={true}
        className="MuiSelect-nativeInput"
        name="productArea"
        onAnimationStart={[Function]}
        onChange={[Function]}
        required={false}
        tabIndex={-1}
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
